readme_renderer/rst.py

```diff
@@ -15,13 +15,45 @@
 
 import io
 
+from docutils import utils
 from docutils.core import publish_parts
 from docutils.writers.html4css1 import HTMLTranslator, Writer
 from docutils.utils import SystemMessage
 
 from .clean import clean
 
 
+def extract_extension_options(field_list, option_spec):
+    """
+    Overrides `utils.extract_extension_options` and inlines
+    `utils.assemble_option_dict` to make it ignore unknown options passed to
+    directives (i.e. ``:caption:`` for ``.. code-block:``).
+    """
+
+    dropped = set()
+    options = {}
+    for name, value in utils.extract_options(field_list):
+        convertor = option_spec.get(name)
+        if name in options or name in dropped:
+            raise utils.DuplicateOptionError('duplicate option "%s"' % name)
+
+        # silently drop unknown options as long as they are not duplicates
+        if convertor is None:
+            dropped.add(name)
+            continue
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏If I'm reading this correctly, this would drop _any_ unknown options, not only the one surfaced as the `code-block` `caption`, correct?

Does this increase the possibility of errors being introduced for other known options? For example, if I [mistyped the `:header:` option to `:headers:`](https://github.com/pypa/readme_renderer/blame/d821b4466457949aa5f61f951d236fcf5ca170da/tests/fixtures/test_rst_caption.rst#L3-L4), I get no warning/error due to this change, and the headers will be missing from the rendered output.

Would it be better to provide this function a specific list of known options we want to skip and thus keep the scope of the override narrow, at least until [doctuils implements `caption` for `code-block`](https://sourceforge.net/p/docutils/feature-requests/62/)?